src/js/models/geocoder/GeocoderSearch.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+define(
+  [
+    'underscore',
+    'backbone',
+    'models/geocoder/GoogleMapsGeocoder',
+    'models/geocoder/GoogleMapsAutocompleter',
+    'models/geocoder/GeocodedLocation',
+    'models/geocoder/Prediction',
+  ],
+  (
+    _,
+    Backbone,
+    GoogleMapsGeocoder,
+    GoogleMapsAutocompleter,
+    GeocodedLocation,
+    Prediction,
+  ) => {
+    /**
+    * @class GeocoderSearch
+    * @classdes GeocoderSearch interfaces with various geocoding and location
+    * searching services.
+    * @classcategory Models/Geocoder
+    */
+    const GeocoderSearch = Backbone.Model.extend({
+      defaults() {
+        return {
+          googleMapsAutocompleter: new GoogleMapsAutocompleter(),
+          googleMapsGeocoder: new GoogleMapsGeocoder(),
+        };
+      },
+
+      /**
+       * Convert a Google Maps Place ID into a list geocoded objects that can be
+       * displayed in the map widget.
+       * @param {string} newQuery - The user's input search query.
+       * @returns {Prediction[]} An array of places that could be the result the
+       * user is looking for. Most often this comes in five or less results.
+       */
+      async autocomplete(newQuery) {
+        return this.getPredictionsFromResults(
+          await this.get('googleMapsAutocompleter').autocomplete(newQuery)
+        );
+      },
+
+      /**
+       * Helper function that converts a Google Maps Autocomplete API result
+       * into a useable Prediction model.
+       * @param {Object[]} List of Google Maps Autocomplete API results.
+       * @returns {Prediction[]} List of corresponding predictions.
+       */
+      getPredictionsFromResults(results) {
+        return results.map(result => new Prediction({
+          description: result.description,
+          googleMapsPlaceId: result.place_id,
+        }));
+      },
+
+      /**
+       * Convert a Google Maps Place ID into a list geocoded objects that can be
+       * displayed in the map widget.
+       * @param {string} placeId - Google Maps Place ID that uniquely identifies
+       * a place in the Google Maps API.
+       * @returns {GeocodedLocation[]} An array of locations with an associated
+       * bounding box. According to Google Maps API this should most often be a
+       * single value, but could potentially be many.
+       */
+      async geocode(placeId) {
```

Comment:
Can do

Reply:
Hm I see this resolved but no commit yet. Forgot to push?

Reply:
It's there, https://github.com/NCEAS/metacatui/pull/2291/commits/7dacf8eb4ae4509cfbbcb06b53137c6e192cbcde

Reply:
oh okay I see the disconnect- I think of GeocoderSearch as the layer of abstraction from the autocomplete and geocode service, making future changes easier if we ever want to move away from GMap. so I think here it probably shouldn't access prediction.placeId, since it's a GMap only concept? Can it just pass the prediction along and let the google service wrapper do the unpacking?

Reply:
I agree that GeocoderSearch is the abstraction layer, anywhere GeocoderSearch is used doesn't need to know about Google Maps. GeocoderSearch _does_ need to know about Google Maps though, so having it access the Places ID is okay. In any case where we change the geocoder backend later, GeocoderSearch will need to be updated. I do think your request to pass in a Prediction here helps in that case, since the call site doesn't need to be updated if we ever change geocoder backend.

Reply:
hmm I think knowing which service to use (google maps), and knowing the service implementation (.placeId) are different. imo, the sole purpose of GeocoderSearch is to decide which service to use. ideally, we just swap out new GoogleMapsGeocoder() for new BingMapsGeocoder() and expect it work OOTB.

that said, I'm only arguing for the principles at this point- I know the difference is very very minor. I'll leave it up to you since we've agreed that GeocoderSearch is the abstraction layer.

Reply:
Thanks for the explanation, updated